### ISSUE TITLE
Newly created marking definitions are not propagated to access control checks

### ISSUE THREAD

**Issue by wattsecs** [LuatixHQ/opencti#7391]
## Description

Newly created marking definitions are not propagated to access control checks in `opencti-platform/opencti-graphql/src/database/middleware.js` until the platform is restarted.

## Environment

1. OS (where OpenCTI server runs): Alpine Docker container
2. OpenCTI version: 6.1.11
3. OpenCTI client: python

## Reproducible Steps

Steps to create the smallest reproducible scenario:
1. register connector, where connector user role does not have bypass, user group has `Automatically authorize this group to new marking definition` enabled
2. connector creates marking definition M
3. connector pushes object O associated to marking definition M
4. ingestion fails with `User trying to create the data has missing markings`
5. restart platform container
6. connector pushes object O associated to marking definition M
7. ingestion succeeds

## Expected Output

Object O is ingested into the platform and marked with M, without having to restart the platform.

## Actual Output

Ingestion fails with `User trying to create the data has missing markings` if the platform is not restarted between marking creation and object ingestion.

## Additional information

Newly created markings are not listed in user allowed markings in `opencti-platform/opencti-graphql/src/database/middleware.js`. They only appear after platform restart.

A similar issue occurs when creating a marking through the GUI:

1. create marking definition M through GUI
2. try to add marking M to an entity
3. marking M is not available for selection
4. refresh page
5. marking M is now available for selection

**Comment by helene-nguyen** [comment]
Hi @jonaskon, I've reproduced the issue above in your additional information and you're right, it is not the behavior expected when you create a new marking.

Concerning the connector that creates a new marking, I think I need more informations about that. 
Could you give us more details about how you have generated the new marking? And how did you add it into your code to get the error message `User trying to create the data has missing markings` ?

**Comment by wattsecs** [comment]
Thank you for investigating.

The new markings are created using the API, but also sent along with the bundle. I am aware that this directly creates the marking in the platform and that sending it in the bundle is redundant in that sense.

The implementation is something like this:
```python
# create marking through API

marking = self.api.marking_definition.create(
	definition_type="placeholder_type",
	definition="placeholder_definition",
)

...

# create corresponding stix2 object

marking = stix2.MarkingDefinition(
	definition_type="statement",
	definition={"statement": "custom"},
	allow_custom=True,
	x_opencti_definition_type=marking["definition_type"],
	x_opencti_definition=marking["definition"],
	x_opencti_color=marking["x_opencti_color"],
)

...

# mark object(s)

some_stix2_object["object_marking_refs"] = [marking["id"]]

# create bundle with object(s) and marking

bundle = self.helper.stix2_create_bundle([marking, some_stix2_object])

# send bundle

self.helper.send_stix2_bundle(bundle)
```

I also tested creating the marking object locally and sending it with the bundle, i.e. not creating the marking on the platform directly through the API, and the issue is the same. The marking object in the bundle gets created, but the object that is marked with this marking does not.

Both implementations work fine, as long as the connector is registered with a user that has the `Bypass all capabilities` capability.

**Comment by BCS251** [comment]
Hello! I'm stepping in to provide some additional context. I encountered the same error with the Sekoia connector by default. When querying GraphQL for the missing marking (TLP: CLEAR) and identity (SEKOIA), the responses returned positive ( no markings were missing ) . Following Jonas's recommendation, I assigned bypass all capabilities to the SEKOIA user. However, despite receiving the intended threat intelligence, the error still persists. in the Ingestion tab.
